examples/ignition-kargs

```diff
@@ -0,0 +1,47 @@
+#!/bin/bash
+
+set -euxo pipefail
+
+# Mount /boot. Note that we mount /boot but we don't unmount it because we
+# are run in a systemd unit with MountFlags=slave so it is unmounted for us.
+bootmnt=/mnt/boot_partition
+mkdir -p ${bootmnt}
+bootdev=/dev/disk/by-label/boot
+mount -o rw ${bootdev} ${bootmnt}
+grubcfg="${bootmnt}/grub/grub.cfg"
+
+kernelopts="$(grep kernelopts= <$grubcfg)"
+
+while [[ $# -gt 0 ]]
+do
+key="$1"
+
+case $key in
+    --should-exist)
+    arg="$2"
+    # don't repeat the arg
+    if [[ ! " ${kernelopts[*]} " =~ " ${arg} " ]]; then
+        kernelopts="$kernelopts $arg "
+    fi
+    shift
+    shift
+    ;;
+    --should-not-exist)
+    arg="$2"
+    kernelopts="$(echo "$kernelopts" | sed "s,$2,,g")"
+    shift
+    shift
+    ;;
+    *)
+    echo "Unknown option"
+    exit 1
+    ;;
+esac
+done
+
+# only apply the changes & reboot if changes have been made
+if [[ ! "$kernelopts" == "$(grep kernelopts= <$grubcfg)" ]]; then
+    awk "{sub(/kernelopts=.*/,\"$kernelopts\"); print}" $grubcfg > tmp && mv tmp $grubcfg
+
+    reboot
```

Comment:
At a minimum, if we're not deferring reboots, this should probably be `systemctl reboot`.